yapapi/config.py

```diff
@@ -0,0 +1,54 @@
+from dataclasses import dataclass, field
+from functools import partial
+import os
+from typing import Optional
+
+
+class MissingConfiguration(Exception):
+    def __init__(self, key: str, description: str):
+        self._key = key
+        self._description = description
+
+    def __str__(self):
+        return f"Missing configuration for {self._description}. Please set env var {self._key}."
+
+
+@dataclass
+class ApiConfig:
+    """
+    Yagna low level API configuration
+    Attributes:
+        subnet_tag: use only providers in the subnet with the subnet_tag name.
```

Comment:
I don't think `subnet_tag` should be part of `ApiConfig`, as it doesn't have anything to do with the yagna API itself... 

and I also have doubts about `payment_driver` and `payment_network` parameters... they are also not part of the configuration of yagna's API -> they pertain to the configuration of the payment layer and as such, should also, most likely, not be included in the configuration of the rest API... 

imvho the only pieces that should be included here are:
* `api_url`
* `market_url`
* `payment_url`
* `net_url`
* `activity_url`

and additionally, 
* `app_key` 

plus, the docs should make it clear that if `market/payment/net/activity` are not provided, they'll be constructed from the base `api_url`...